CMakeLists.txt

```diff
@@ -43,1343 +88,243 @@ add_executable(
+set_property(TARGET v8-i18n-support PROPERTY EXCLUDE_FROM_ALL ON)
+
+list(APPEND i18n-sources
+  ${PROJECT_SOURCE_DIR}/v8/src/builtins/builtins-intl.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/intl-objects.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-break-iterator.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-collator.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-date-time-format.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-display-names.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-list-format.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-locale.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-number-format.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-plural-rules.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-relative-time-format.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-segment-iterator.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/objects/js-segmenter.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/runtime/runtime-intl.cc
+  ${PROJECT_SOURCE_DIR}/v8/src/strings/char-predicates.cc)
+
+target_sources(v8-i18n-support PRIVATE ${i18n-sources})
+target_compile_options(v8-i18n-support PRIVATE ${disable-exceptions})
+target_link_libraries(v8-i18n-support PRIVATE v8_torque_generated)
+
+target_include_directories(v8-i18n-support
+  PRIVATE
+    ${D}
+    ${D}/src/objects
+    ${D}/include)
+
+file(GLOB api-sources CONFIGURE_DEPENDS v8/src/api/*.cc)
+file(GLOB asmjs-sources CONFIGURE_DEPENDS v8/src/asmjs/*.cc)
+file(GLOB ast-sources CONFIGURE_DEPENDS v8/src/ast/*.cc)
+file(GLOB builtin-sources CONFIGURE_DEPENDS v8/src/builtins/*.cc)
+file(GLOB codegen-sources CONFIGURE_DEPENDS v8/src/codegen/*.cc)
+file(GLOB debug-sources CONFIGURE_DEPENDS v8/src/debug/*.cc)
+file(GLOB deoptimizer-sources CONFIGURE_DEPENDS v8/src/deoptimizer/*.cc)
+file(GLOB diagnostic-sources CONFIGURE_DEPENDS v8/src/diagnostics/*.cc)
+file(GLOB execution-sources CONFIGURE_DEPENDS v8/src/execution/*.cc)
+file(GLOB extensions-sources CONFIGURE_DEPENDS v8/src/extensions/*.cc)
+file(GLOB interpreter-sources CONFIGURE_DEPENDS v8/src/interpreter/*.cc)
+file(GLOB json-sources CONFIGURE_DEPENDS v8/src/json/*.cc)
+file(GLOB logging-sources CONFIGURE_DEPENDS v8/src/logging/*.cc)
+file(GLOB heap-sources CONFIGURE_DEPENDS v8/src/heap/*.cc)
+file(GLOB numbers-sources CONFIGURE_DEPENDS v8/src/numbers/*.cc)
+file(GLOB object-sources CONFIGURE_DEPENDS v8/src/objects/*.cc)
+file(GLOB parsing-sources CONFIGURE_DEPENDS v8/src/parsing/*.cc)
+file(GLOB profiler-sources CONFIGURE_DEPENDS v8/src/profiler/*.cc)
+file(GLOB regexp-sources CONFIGURE_DEPENDS v8/src/regexp/*.cc)
+file(GLOB runtime-sources CONFIGURE_DEPENDS v8/src/runtime/*.cc)
+file(GLOB snapshot-sources CONFIGURE_DEPENDS v8/src/snapshot/*.cc)
+file(GLOB strings-sources CONFIGURE_DEPENDS v8/src/strings/*.cc)
+file(GLOB utils-sources CONFIGURE_DEPENDS v8/src/utils/*.cc)
+file(GLOB wasm-sources CONFIGURE_DEPENDS v8/src/wasm/*.cc)
+file(GLOB wasm-baseline-sources CONFIGURE_DEPENDS v8/src/wasm/baseline/*.cc)
+file(GLOB zone-sources CONFIGURE_DEPENDS v8/src/zone/*.cc)
```

Comment:
One thing I'm not completely clear about is how you filter out files for different architectures. E.g. when compiling for x64, file names ending in -x64.{cc,h} should be included, but not arm, ia32, s390, etc.

Reply:
If you look below you'll see I can manually remove some names. If we need to do it on a filtering basis, you can use the 

`list(FILTER ${var}-sources EXCLUDE REGEX ".*-x64[.]cc$")` command. It's not as clean as a glob, but it does allow us to do stuff like that. We can then glob them specifically.